iroh-net/src/endpoint.rs

```diff
@@ -853,14 +853,12 @@ impl Endpoint {
             endpoint,
             cancel_token,
             ..
-        } = self;
+        } = &self;
```

Comment:
to actually change the behaviour of this function, this needs to take `&self` as the argument, so the function signature should become

```rust
pub async fn close(&self, error_code: VarInt, reason: &[u8]) -> Result<()>
 ```

Reply:
Thanks for the review!

Reply:
```suggestion
        } = self;
```
I don't think this change is needed